pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/store/SegmentDirectory.java

```diff
@@ -181,24 +180,28 @@ public abstract PinotDataBuffer getIndexFor(String column, ColumnIndexType type)
 
     public abstract boolean hasIndexFor(String column, ColumnIndexType type);
 
-    public SegmentDirectory toSegmentDirectory() {
-      return SegmentDirectory.this;
-    }
-
-    public abstract String toString();
-
-    public PinotDataBuffer getStarTreeIndex()
```

Comment:
fyi, `getStarTreeIndex ` and `getStarTreeIndexMap` were added very recenlty by https://github.com/apache/pinot/pull/9828 but they didn't abstract the access to ST index very cleanly. This PR improves on this.